### PR TITLE
Feature reducers

### PR DESCRIPTION
**Pull request by baparham**
- [x] geomReduce (@baparham)
- [x] need to redo the JS comment for geomReduce (@armantorkzaban)
--- 
- [x] propReduce (@armantorkzaban)
- [x] propReduce tests (@armantorkzaban)

---

- [x] featureReduce (@armantorkzaban)
- [x] featureReduce tests (@armantorkzaban)

---

- [x] flattenReduce (@armantorkzaban)
- [x] flattenReduce tests (@armantorkzaban)

---

- [x] coordReduce (@armantorkzaban)
- [x] coordReduce tests (@armantorkzaban)

### CODE REVIEW

**Comment by lukas-h** [comment]
#48 is merged, so nothing stands in the way of implementing the functions! 😄 

CC @armantorkzaban

**Comment by armantorkzaban** [comment]
I am working on this one.

File: lib/src/meta.dart [hunk 1]
```diff
@@ -158,8 +168,8 @@ class _ShortCircuit {
   _ShortCircuit();
 }
 
-/// Iterates over each geometry in [geoJSON], calling [callback] on each
-/// iteration. Similar to Iterable.forEach()
+/// Iterate over each geometry in [geoJSON], calling [callback] on each
+/// iteration. Similar to [List.forEach()]
```
Comment (lukas-h) [inline lib/src/meta.dart:172]: in this case, I think Iterable.forEach() was better

File: lib/src/meta.dart [hunk 2]
```diff
@@ -282,7 +369,7 @@ typedef FeatureEachCallback = dynamic Function(
     Feature currentFeature, num featureIndex);
```
Comment (armantorkzaban) [inline lib/src/meta.dart:369]: Should it not be int?

**Comment by lukas-h** [comment]
@armantorkzaban 

We have to make a technical decision about the types of the reducer functions. I thought of two ways to handle it.
1. `dynamic` types:
    The previousValue will be initialized if initialValue==null like in the original JS version and also like in Arman’s and Brad’s initial versions. We will drop the generic types to not limit it to geodata-related data types.

3. generic types: Very clean, but we break a little bit with Turf.js-compatibility: We don’t have the if-clause to initialize previousValue if initialValue==null. I’ve implemented that in the latest commit to this branch.

Let’s decide together!

**Comment by lukas-h** [comment]
We now decided for a different approach, where we don't lose any type information and are entirely compatible with TurfJS!
But in this version, we added another type check to the functions:
 
```dart
      if (previousValue == null && featureIndex == 0 && currentGeometry is T) {
        previousValue = currentGeometry?.clone() as T;
      } else {
```

**Comment by lukas-h** [comment]
@armantorkzaban For the unit tests, please look at geomReduce, Brad did a great job there. I added some lines to test the behavior with `dynamic` types

**Review by lukas-h — APPROVED** [review]

**Pull request merged**